cpp/src/arrow/util/meson.build

```diff
@@ -92,3 +92,242 @@ configure_file(
+        'type_fwd.h',
+        'type_traits.h',
+        'ubsan.h',
+        'union_util.h',
+        'unreachable.h',
+        'uri.h',
+        'utf8.h',
+        'value_parsing.h',
+        'vector.h',
+        'visibility.h',
+        'windows_compatibility.h',
+        'windows_fixup.h',
+    ],
+    subdir: 'arrow/util',
+)
+
+utility_test_srcs = [
+    'align_util_test.cc',
+    'atfork_test.cc',
+    'byte_size_test.cc',
+    'byte_stream_split_test.cc',
+    'cache_test.cc',
+    'checked_cast_test.cc',
+    'compression_test.cc',
+    'decimal_test.cc',
+    'float16_test.cc',
+    'fixed_width_test.cc',
+    'formatting_util_test.cc',
+    'key_value_metadata_test.cc',
+    'hashing_test.cc',
+    'int_util_test.cc',
+    'io_util_test.cc',
+    'iterator_test.cc',
+    'list_util_test.cc',
+    'logger_test.cc',
+    'logging_test.cc',
+    'math_test.cc',
+    'queue_test.cc',
+    'range_test.cc',
+    'ree_util_test.cc',
+    'reflection_test.cc',
+    'rows_to_batches_test.cc',
+    'small_vector_test.cc',
+    'span_test.cc',
+    'stl_util_test.cc',
+    'string_test.cc',
+    'tdigest_test.cc',
+    'test_common.cc',
+    'time_test.cc',
+    'tracing_test.cc',
+    'trie_test.cc',
+    'uri_test.cc',
+    'utf8_util_test.cc',
+    'value_parsing_test.cc',
+]
+
+if host_machine.system() == 'windows'
+    # This manifest enables long file paths on Windows 10+
+    # See https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#enable-long-paths-in-windows-10-version-1607-and-later
+    if cpp_compiler.get_id() == 'msvc'
+        utility_test_sources += ['io_util_test.manifest']
+    else
+        utility_test_sources += ['io_util_test.rc']
+    endif
+endif
+
+exc = executable(
+    'arrow-utility-test',
+    sources: utility_test_srcs,
+    dependencies: [arrow_dep, filesystem_dep, gtest_dep, gmock_dep],
+    link_with: [arrow_test_lib],
+    implicit_include_directories: false,
+)
+test('arrow-utility-test', exc)
+
+util_tests = {
+    'arrow-async-utility-test': {
+        'sources': [
+            'async_generator_test.cc',
+            'async_util_test.cc',
+            'test_common.cc',
+        ],
+    },
+    'arrow-bit-utility-test': {
+        'sources': [
+            'bit_block_counter_test.cc',
+            'bit_util_test.cc',
+            'rle_encoding_test.cc',
+        ],
+    },
+    'arrow-threading-utility-test': {
+        'sources': [
+            'cancel_test.cc',
+            'counting_semaphore_test.cc',
+            'future_test.cc',
+            'task_group_test.cc',
+            'test_common.cc',
+            'thread_pool_test.cc',
+        ],
+    },
+    'arrow-crc32-test': {
+        'sources': ['crc32_test.cc'],
+        'dependencies': [filesystem_dep],
+    },
+}
+
+if needs_tests
+    foreach key, val : util_tests
+        exc = executable(
+            key,
+            sources: val['sources'],
+            dependencies: [arrow_test_dep, val.get('dependencies', [])],
```

Comment:
Do we need `gtest_main_dep` here?

Reply:
`arrow_test_dep` already provides `gtest_main_dep` transitively

Reply:
Ah, `arrow_test_dep` doesn't link to only `libarrow_testing.so`.

Then can we remove `filesystem_dep` from `arrow-crc32-test` dependencies?

Reply:
Ah nice catch - yea we can do that for now